### DISCUSSION TITLE
Expected resolved value to be an object but got "array"

### DISCUSSION THREAD

**Discussion by dhiabenjaber** [thecodingmachine/graphqlite#461] [Q&A] [answered]
i have this isue "Expected resolved value to be an object but got "array"" i want to resolve a field that returns an array i do @return NameClass[] and i got this error anyone has an idea

**Comment by oojacoboo** [comment]
Make sure the return annotation typing is a FQCN or included in a `use` statement.  Does that help?  It sounds like you’re doing it correctly.  If you’re still having issues, please share more code.

**Comment by dhiabenjaber** [comment]
/**
     * @Field
     * @return NearestShops[]
     */
    public function getNearestShops():array
    {
        $nearestShops = new NearestShops($this->shopId, $this->latitude, $this->longitude,$this->url,$this->name,$this->id);
        $arr=$nearestShops->getNearestShops();
        return $arr;


    }
this is the field in the model class
and this is the class model that i use it in the previous field
 * @Type
 */

class NearestShops 
{
    public function __construct($shopId, $latitude, $longitude,$id,$name,$url )
    {
        $this->shopId=$shopId;
        $this->latitude=$latitude;
        $this->longitude=$longitude;
        $this->url=$url;
        $this->id=$id;
        $this->name=$name;


    }
    /**
     * Porte l'url de picture du magasin.
     * @var string
     * @Field()
     */
    public function id():?string
    {
        return $this->id;
    }
    /**
     * Porte l'url de picture du magasin.
     * @var string
     * @Field()
     */
    public function name():?string
    {
        return $this->name;
    }
    /**
     * Porte l'url de picture du magasin.
     * @var string
     * @Field()
     */
    public function url():?string
    {
        return $this->url;
    }

    /**
     * @return array
     */
    public function getNearestShops()
    {
        $shopDBResource = new \Kawa\Modules\Shop\Models\GraphqlModels\Resources\ShopDBResource();
        $shopNearestShops = $shopDBResource->getNearestShops($this->shopId, $this->latitude, $this->longitude);
        $nearestShops = array();
        foreach ($shopNearestShops->getItems() as $nearShop) {
            $nearShopData = $nearShop->getData();
            $nearShopObject = new Shop($nearShopData['shopId']);
            $nearShopObject->load();
            $url = "/magasins/" . $nearShopData['shopId'] . "/" . Helper::filterSanitizeString($nearShopData['name']) . ".html";
            array_push($nearestShops, array('id' => $nearShopData['shopId'], 'name' => $nearShopData['name'], 'url' => $url));

        }
        return $nearestShops;
    }
}
the problem is that the function return an array should i return an array in the constructor ??

**Comment by oojacoboo** [comment]
@dhiabenjaber firstly, case sensitivity matters.  Secondly, I don't see any of your `use` statements in the examples.

**Comment by dhiabenjaber** [comment, marked as answer]
problem fixed here is the code 
/**
     * @Field
     * @return NearestShops[]
     */
    public function nearestShops()
    {
//        $nearestShops = new NearestShops($this->shopId, $this->latitude, $this->longitude,$this->url,$this->name,$this->id);
        $arr=$this->getNearestShops();
        $shopArray = array();
        foreach ($arr as $item) {

            $nearestShop = new NearestShops($this->shopId, $this->latitude, $this->longitude,$item["id"],$this->name,$item["url"]);

            array_push($shopArray, $nearestShop);

        }
        return $shopArray;


    }
